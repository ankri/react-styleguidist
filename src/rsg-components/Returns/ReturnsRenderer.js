import React from 'react';
import PropTypes from 'prop-types';
import Argument from 'rsg-components/Argument';
import Heading from 'rsg-components/Heading';
import Styled from 'rsg-components/Styled';

export const styles = ({ space }) => ({
	root: {
		marginBottom: space[2],
		fontSize: 'inherit',
	},
	headingWrapper: {
		marginBottom: space[0],
	},
});

export function ReturnsRenderer({ classes, returnDocumentation, heading }) {
	if (!returnDocumentation) {
		return null;
	}

	return (
		<div className={classes.root}>
			{heading && (
				<div className={classes.headingWrapper}>
					<Heading level={5}>Returns</Heading>
				</div>
			)}
			<Argument key={returnDocumentation.title} {...returnDocumentation} />
		</div>
	);
}

ReturnsRenderer.propTypes = {
	classes: PropTypes.object.isRequired,
	returnDocumentation: PropTypes.shape({
		title: PropTypes.string.isRequired,
		type: PropTypes.object,
		description: PropTypes.string,
	}).isRequired,
	heading: PropTypes.bool,
};

export default Styled(styles)(ReturnsRenderer);
