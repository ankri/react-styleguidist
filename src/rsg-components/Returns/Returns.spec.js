import React from 'react';
import { ReturnsRenderer, styles } from './ReturnsRenderer';

const props = {
	classes: classes(styles),
};

const returnDocumentation = [
	{
		title: 'Foo',
		description: 'Converts foo to bar',
		type: { name: 'String' },
	},
	{
		title: 'Bar',
		description: 'Converts bar to foo',
		type: null,
	},
	{
		title: 'Baz',
		type: { name: 'SyntheticEvent' },
	},
];

it('renderer should render type and description', () => {
	const actual = shallow(
		<ReturnsRenderer {...props} returnDocumentation={returnDocumentation[0]} />
	);

	expect(actual).toMatchSnapshot();
});

it('renderer should render only description', () => {
	const actual = shallow(
		<ReturnsRenderer {...props} returnDocumentation={returnDocumentation[1]} />
	);

	expect(actual).toMatchSnapshot();
});

it('renderer should render only type', () => {
	const actual = shallow(
		<ReturnsRenderer {...props} returnDocumentation={returnDocumentation[2]} />
	);

	expect(actual).toMatchSnapshot();
});

it('renderer should render heading', () => {
	const actual = shallow(
		<ReturnsRenderer {...props} returnDocumentation={returnDocumentation[0]} heading />
	);

	expect(actual).toMatchSnapshot();
});
