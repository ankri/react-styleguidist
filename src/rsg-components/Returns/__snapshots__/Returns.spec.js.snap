// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renderer should render heading 1`] = `
<div
  className="root"
>
  <div
    className="headingWrapper"
  >
    <Styled(Heading)
      level={5}
    >
      Returns
    </Styled(Heading)>
  </div>
  <Styled(Argument)
    description="Converts foo to bar"
    key="Foo"
    title="Foo"
    type={
      Object {
        "name": "String",
      }
    }
  />
</div>
`;

exports[`renderer should render only description 1`] = `
<div
  className="root"
>
  <Styled(Argument)
    description="Converts bar to foo"
    key="Bar"
    title="Bar"
    type={null}
  />
</div>
`;

exports[`renderer should render only type 1`] = `
<div
  className="root"
>
  <Styled(Argument)
    key="Baz"
    title="Baz"
    type={
      Object {
        "name": "SyntheticEvent",
      }
    }
  />
</div>
`;

exports[`renderer should render type and description 1`] = `
<div
  className="root"
>
  <Styled(Argument)
    description="Converts foo to bar"
    key="Foo"
    title="Foo"
    type={
      Object {
        "name": "String",
      }
    }
  />
</div>
`;
